src/core/control/tools/StrokeHandler.cpp

```diff
@@ -266,11 +268,23 @@ void StrokeHandler::onButtonReleaseEvent(const PositionInputData& pos, double zo
     Layer* layer = page->getSelectedLayer();
 
     UndoRedoHandler* undo = control->getUndoRedoHandler();
-
     undo->addUndoAction(std::make_unique<InsertUndoAction>(page, layer, stroke.get()));
 
-    ToolHandler* h = control->getToolHandler();
+    if (settings->getEmptyLastPageAppend() == EmptyLastPageAppendType::OnDrawOfLastPage) {
+        auto* doc = control->getDocument();
+        doc->lock();
+        if (doc->getPdfPageCount() == 0) {
+            auto currentPage = control->getCurrentPageNo();
+            auto lastPage = doc->getPageCount() - 1;
+            if (currentPage == lastPage) {
+                doc->unlock();
+                control->insertNewPage(currentPage + 1, false);
+            }
+        }
+        doc->unlock();
```

Comment:
You may unlock an already unlocked mutex here. I would suggest this:
```suggestion
        doc->lock();
        auto pdfPageCount = doc->getPdfPageCount();
        doc->unlock();
        if (pdfPageCount == 0) {
            doc->lock();
            auto currentPage = control->getCurrentPageNo();
            auto lastPage = doc->getPageCount() - 1;
            doc->unlock();
            if (currentPage == lastPage) {
                control->insertNewPage(currentPage + 1, false);
            }
        }
```
Similarly for the other occurence.

Reply:
Thank you for your suggestions - they are now applied :)